src/PhpWord/Writer/Word2007/Part/Chart.php

```diff
@@ -265,23 +266,21 @@ private function writeSeries(XMLWriter $xmlWriter, $scatter = false)
                 $this->writeSeriesItem($xmlWriter, 'cat', $categories);
                 $this->writeSeriesItem($xmlWriter, 'val', $values);
 
-                // setting the chart colors was taken from https://github.com/PHPOffice/PHPWord/issues/494
-                if (is_array($colors) && count($colors)) {
-                    // This is a workaround to make each series in a stack chart use a different color
-                    if ($this->options['type'] == 'bar' && stristr($this->options['grouping'], 'stacked')) {
-                        array_shift($colors);
-                    }
-                    $colorIndex = 0;
-                    foreach ($colors as $color) {
+                // check that there are colors
+                if (is_array($colors) && count($colors)>0) {
+                    // assign a color to each value
+                    $valueIndex=0;
```

Comment:
is this workaround for stacked charts not needed anymore?

Reply:
how did you fix this? without the shift for stacked bars the unit test testChartElements in ChartTest fails.

Reply:
I do not alter the array but get the next color using modular computation :
$xmlWriter->writeElementBlock('a:srgbClr', 'val', $colors[$colorIndex++ % count($colors)]);